### PR TITLE
Add stub Ruby commands for all Bash commands.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This gets us pretty similar (but easier to manage) manpage output but much nicer completions etc. for all these commands.

Based on what @ZhongRuoyu did for `brew --repository`.

### CODE REVIEW

File: Library/Homebrew/abstract_command.rb [hunk 1]
```diff
@@ -32,6 +32,12 @@ def command_name
              .delete_suffix("-cmd")
       end
 
+      sig { void }
+      def raise_sh_command_error!
+        raise StandardError, "This command is just here for completions generation. " \
+                             "It's actually defined in `cmd/#{command_name}` instead."
```
Comment (carlocab) [inline Library/Homebrew/abstract_command.rb:38]: You could probably put this into the default definition of `#run` so you can just leave `#run` unimplemented for Ruby stubs.

Comment (ZhongRuoyu) [reply Library/Homebrew/abstract_command.rb:38]: I'd say being more explicit here seems better; to me it doesn't make much sense to stub by default (which also presumes the `.sh` counterpart exists).

Comment (Bo98) [reply Library/Homebrew/abstract_command.rb]: How about:

```rb
# central file
module ShellCommand
  def run
    raise ...
  end
end

# in each cmd
class Update < AbstractCommand
  include ShellCommand

  ...
end
```

Bonus is it works as a metadata marker and could mean we have a test that validates this: e.g. `update.sh` must exist if `Homebrew::Cmd::Update.include?(ShellCommand)`

Comment (MikeMcQuaid) [reply Library/Homebrew/abstract_command.rb]: Can we do this as a follow-up refactoring instead? This PR is an improvement as-is.

**Review by Bo98 — APPROVED** [review]

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**